hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSource.scala

```diff
@@ -723,4 +723,29 @@ class TestCOWDataSource extends HoodieClientTestBase {
     val result = spark.sql("select * from tmptable limit 1").collect()(0)
     result.schema.contains(new StructField("partition", StringType, true))
   }
+
+  @Test
+  def testWriteSmallPrecisionDecimalTable(): Unit = {
```

Comment:
Can we add one more test, where if not for small precision, if user has set hoodie.parquet.writeLegacyFormat.enabled to false, it should not get overridden.

Reply:
It is difficult to get the value of hoodie.parquet.writeLegacyFormat.enabled directly from spark.
add functions test for autoModifyParquetWriteLegacyFormatParameter to cover all scenes.